packages/votemarket/src/oracle/PendleOracleLens.sol

```diff
@@ -0,0 +1,56 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity 0.8.19;
+
+///  Project Interfaces
+import "src/interfaces/IPendleOracle.sol";
+
+/// @notice Oracle Lens contract to read from the Oracle contract and expose data to the Votemarket contract.
+/// @dev It should always revert if the epoch data is not updated.
+contract PendleOracleLens {
+    /// @notice Oracle address.
+    address public immutable oracle;
+
+    /// @notice Thrown when the epoch data is not updated.
+    error STATE_NOT_UPDATED();
+
+    constructor(address _oracle) {
+        oracle = _oracle;
+    }
+
+    /// @notice Checks if an account can claim rewards for a gauge and epoch.
+    /// @param account Account address.
+    /// @param gauge Gauge address.
+    /// @param epoch Epoch number.
+    /// @return bool True if the account can claim, false otherwise.
+    function isVoteValid(address account, address gauge, uint256 epoch) external view returns (bool) {
+        IPendleOracle.VotedSlope memory account_ = IPendleOracle(oracle).votedSlopeByEpoch(account, gauge, epoch);
+        if (account_.lastUpdate == 0) revert STATE_NOT_UPDATED();
```

Comment:
Your call on this, but I’d recommend turning the `lastUpdate == 0` check into a simple return false rather than a revert. That keeps this external view function purely “ask-only,” improves composability, and avoids forcing callers to handle an exception for a, in my point of view, valid “no vote yet” scenario.

Again, your call on this. I just tend to favor simpler function and smart-contract interaction these days.